internal/cloud_posture/rules_data_source.go

```diff
@@ -0,0 +1,426 @@
+		)
+
+		return nil, diags
+	}
+
+	queryPayload := queryResp.GetPayload()
+
+	if err = falcon.AssertNoError(queryPayload.Errors); err != nil {
+		diags.AddError(
+			"Failed to query rules",
+			fmt.Sprintf("Failed to query rules: %s", err.Error()),
+		)
+		return nil, diags
+	}
+
+	if len(queryPayload.Resources) == 0 {
+		diags.AddWarning(
+			"No Rules Found",
+			"The query returned no rules. Please check your filter criteria.",
+		)
+		return nil, diags
+	}
+
+	ruleParams := cloud_policies.GetRuleParams{
+		Context: ctx,
+		Ids:     queryPayload.Resources,
+	}
+
+	getRulesResp, err := r.client.CloudPolicies.GetRule(&ruleParams)
+	if err != nil {
+
+		if !strings.Contains(err.Error(), "rule resource doesn't exist") {
+			diags.AddError(
+				"Failed to get rule",
+				fmt.Sprintf("Failed to get rule: %s", err),
+			)
+		}
+		return nil, diags
+	}
+
+	getRulesPayload := getRulesResp.GetPayload()
+
+	if err = falcon.AssertNoError(getRulesPayload.Errors); err != nil {
+		diags.AddError(
+			"Failed to get rules",
+			fmt.Sprintf("Failed to get rule: %s", err.Error()),
+		)
+		return nil, diags
+	}
+
+	for _, resource := range getRulesPayload.Resources {
+		alertInfo := convertAlertRemediationInfoToTerraformState(resource.AlertInfo)
+
+		rule := cloudPostureRulesDataSourceRuleModel{
+			UUID:            types.StringValue(*resource.UUID),
+			AlertInfo:       alertInfo,
+			Description:     types.StringPointerValue(resource.Description),
+			AutoRemediable:  types.BoolPointerValue(resource.AutoRemediable),
+			Domain:          types.StringPointerValue(resource.Domain),
+			Logic:           types.StringValue(resource.Logic),
+			Name:            types.StringPointerValue(resource.Name),
+			ParentRuleID:    types.StringValue(resource.ParentRuleShortUUID),
+			CloudPlatform:   types.StringValue(resource.Platform),
+			CloudProvider:   types.StringPointerValue(resource.Provider),
+			RemediationInfo: types.StringPointerValue(resource.RuleLogicList[0].RemediationInfo),
+			ResourceType:    types.StringPointerValue(resource.ResourceTypes[0].ResourceType),
+			Severity: func() types.Int32 {
+				if resource.Severity != nil {
+					return types.Int32Value(int32(*resource.Severity))
+				}
+				return types.Int32Null()
+			}(),
+			Subdomain: types.StringPointerValue(resource.Subdomain),
+		}
+
+		for _, control := range resource.Controls {
+			rule.Controls = append(rule.Controls, struct {
+				Authority types.String `tfsdk:"authority"`
+				Code      types.String `tfsdk:"code"`
+			}{
+				Authority: types.StringPointerValue(control.Authority),
+				Code:      types.StringPointerValue(control.Code),
+			})
+		}
+
+		rule.AttackTypes = types.SetValueMust(types.StringType, []attr.Value{})
+		for _, attackType := range resource.AttackTypes {
+			rule.AttackTypes, diags = types.SetValue(types.StringType, append(rule.AttackTypes.Elements(), types.StringValue(attackType)))
+			if diags.HasError() {
+				return nil, diags
+			}
+		}
```

Comment:
you should be able to do this instead


```suggestion
        attackTypeSet, diags := types.SetValueFrom(ctx, types.StringType, resource.AttackTypes)
        if diags.HasError() {
            return nil, diags
        }
```